File: 58.py
#!/usr/bin/python

p = [False,False,True,True,False,True]

def isP(n):
    for i in xrange(2,n):
        if i * i > n:
            return True
        if n % i == 0:
            return False

def isPrime(n):
    if n < 0:
        return False
    if len(p) > n:
        return p[n]
    for i in xrange(len(p), n+1):
        p.append(isP(i))
    return p[n]

step = 2
x = 1
cnt = 0 
i = 0

while True:
    i += 1
    for j in xrange(4):
        x += step
        cnt += isPrime(x)
    step += 2
    if cnt * 10 < 4 * i + 1:
        print cnt, 4 * i + 1, i * 2 + 1
        exit(0)

File: 58.c
#include <stdio.h>
#include <stdlib.h>

int isPrime(int x) {
  if (x < 2)
    return 0;
  for (int i = 2; i * i <= x; i++) {
    if (x % i == 0) {
      return 0;
    }
  }
  return 1;
}

int main(void) {
  int step = 2;
  int x = 1;
  int cnt = 0;
  int i = 0;

  while (1) {
    i += 1;
    for (int i = 0; i < 4; i++) {
      x += step;
      cnt += isPrime(x);
    }
    step += 2;
    if (cnt * 10 < 4 * i + 1) {
      printf("%d %d %d\n", cnt, 4 * i + 1, i * 2 + 1);
      return 0;
    }
  }
  return 1;
}
